src/test/java/org/broadinstitute/hellbender/engine/GatkToolIntegrationTest.java

```diff
@@ -91,4 +102,66 @@ public void testSharding() {
         Assert.assertTrue(Files.exists(Paths.get(firstShard + FileExtensions.COMPRESSED_VCF_INDEX)));
         Assert.assertTrue(Files.exists(Paths.get(secondShard + FileExtensions.COMPRESSED_VCF_INDEX)));
     }
+
+    @CommandLineProgramProperties(summary = "testTool which emits specific variants",
+            oneLineSummary = "Test tool",
+            programGroup = TestProgramGroup.class)
+    public static class VariantEmitter extends GATKTool{
+        @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+        File output;
+
+        @Override
+        public void traverse() {
+            //nope
+        }
+
+        @Override
+        public void onTraversalStart() {
+            try(final VariantContextWriter vcfWriter = createVCFWriter(output)){
+                vcfWriter.writeHeader(new VCFHeader());
+                final VariantContextBuilder vcb = new VariantContextBuilder();
+                vcb.alleles("AAAAAA", "A").chr("1");
+
+                vcfWriter.add(vcb.start(10).stop(15).make());
+                vcfWriter.add(vcb.start(100).stop(105).make());
+                vcfWriter.add(vcb.start(1000).stop(1005).make());
+                vcfWriter.add(vcb.start(10000).stop(10005).make());
+
+                vcb.chr("2");
+                vcfWriter.add(vcb.start(20).stop(25).make());
+                vcfWriter.add(vcb.start(200).stop(205).make());
+                vcfWriter.add(vcb.start(2000).stop(2005).make());
+                vcfWriter.add(vcb.start(20000).stop(20005).make());
+            }
+        }
+    }
+
+    @DataProvider
+    public Object[][] getIntervalsAndOverlapMode(){
```

Comment:
Maybe better to check the actual intervals retained, rather than just the total number?